### DISCUSSION TITLE
About Firecracker Usage..

### DISCUSSION THREAD

**Discussion by Jeffr-K** [firecracker-microvm/firecracker#4479] [Q&A] [answered]
Hello, Firecracker heros. I am a junior software developer.

I am looking for a micro virtual environment to specify specific works in my project..

I have some critical issue. An user must have one microVM contained react(frontend), server, database. 

The firecracker can resolve this issue focused on VM capacity?

I am sorry, I am also a newbie in devops section too.

<img width="776" alt="스크린샷 2024-02-29 오후 7 20 38" src="https://github.com/firecracker-microvm/firecracker/assets/89901793/5e1ce7ff-7ea5-45c7-bda3-9a78af4a024b">

**Comment by cipriancraciun** [comment, marked as answer]
I didn't quite get what you are asking about.  (Some sentences seem to have been poorly translated.)

However, based on your diagram, I think you are asking if Firecracker supports running in the same "microVM" both the frontend / backend and a database belonging to the same application?

If so, the question is "yes", you can use Firecracker this way.

Take into account that Firecracker is a virtualization technology (focusing on certain aspects) that just "runs" a Linux-based VM.  Thus it doesn't care whatever you deploy in that VM.

----

However, perhaps a more important question is why do you want to use Firecracker?  Is security (sandboxing), start-up latency, or multiple executions scalability an important aspect for your use-case?  Or do you just want to have isolated environments to test various aspects of your application?

Because if you only need isolation, perhaps plain-old-containers (via Podman or Docker) would better suit your use-case.

(With Firecracker you need to provide the VM image yourself, as a raw disk image, and there isn't a built-in tool to generate that image.  Also you need to manage the startup / teardown of your services, etc.  As stated, Firecracker is just a virtualization building-block, not an featurefull product / platform.)

**Reply by Jeffr-K** [reply]
Thanks for your reply.

I guess I didn't understand the "firecracker" itself. But your answer was very good to me.
